exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -748,6 +771,22 @@ public ParquetTableMetadataDirs(List<String> directories) {
       return directories;
     }
 
+    /** If directories list contains relative paths, update it to absolute ones
+     * @param baseDir base parent directory
+     */
+    @JsonIgnore public void updateRelativePaths(Path baseDir) {
+      if (!directories.isEmpty()) {
+        // It is enough to check the first path to decide if updating needed
+        if (!new Path(directories.get(0)).isAbsolute()) {
```

Comment:
It is possible to replace String with Path for directories paths due to implementing custom `JsonSerializer<Path>` and `JsonDeserializer<Path>`. But then it will be necessary to convert every `Path` from lists back into `String`, because a String paths are used in a lot of places: `FileSelection`, `Metadata`, `ParquetGroupScan`, `ReadEntryWithPath`, `FileWork`, `FormatSelection`, `FormatPlugin`,  `PartitionLocation`and so on.

I am totally agree with replacing `String` with `Path` requirement. But it should be done not only for parquet and in context of separate jira. I am going to create it.